$NetBSD: patch-configure.ac,v 1.2 2025/06/02 04:37:03 jnemeth Exp $

1. Don't clear CPPFLAGS/LDFLAGS and impose /usr/pkg or /usr/local;
   pkgsrc will handle its own prefix, which may not be /usr/pkg in
   which case asterisk should absolutely never look at /usr/pkg.
2. Ditto.
3. Standard poll is in <poll.h> -- check for that, not <sys/poll.h>.
4. Check for strftime_l.
5. Standard poll is in <poll.h> -- check for that, not <sys/poll.h>.

--- configure.ac.orig	2025-05-22 15:57:26.000000000 +0000
+++ configure.ac
@@ -65,13 +65,9 @@ esac
 case "${host_os}" in
      netbsd*)
      ac_default_prefix=/usr/pkg
-     CPPFLAGS=-I/usr/pkg/include
-     LDFLAGS=-L/usr/pkg/lib
      ;;
      dragonfly*|freebsd*)
      ac_default_prefix=/usr/local
-     CPPFLAGS=-I/usr/local/include
-     LDFLAGS=-L/usr/local/lib
      ;;
      openbsd*)
      ac_default_prefix=/usr/local
@@ -83,8 +79,6 @@ case "${host_os}" in
            astmandir=/usr/share/man
         fi
      fi
-     CPPFLAGS=-I/usr/local/include
-     LDFLAGS=-L/usr/local/lib
      ;;
      darwin*)
      ac_default_prefix=/usr/local
@@ -828,7 +822,7 @@ AC_CHECK_HEADERS([xlocale.h])
 
 AC_CHECK_HEADERS([winsock.h winsock2.h])
 
-AC_CHECK_HEADER([sys/poll.h],
+AC_CHECK_HEADER([poll.h],
 	 [],
      AC_DEFINE([AST_POLL_COMPAT], 1, [Define to 1 if internal poll should be used.]))
 
@@ -880,7 +874,7 @@ AC_FUNC_STRNLEN
 AC_FUNC_STRTOD
 AC_FUNC_UTIME_NULL
 AC_FUNC_VPRINTF
-AC_CHECK_FUNCS([asprintf atexit closefrom dup2 eaccess endpwent euidaccess ffsll ftruncate getcwd gethostbyname gethostname getloadavg gettimeofday glob ioperm inet_ntoa isascii memchr memmove memset mkdir mkdtemp munmap newlocale pipe2 ppoll putenv re_comp regcomp select setenv socket strcasecmp strcasestr strchr strcspn strdup strerror strlcat strlcpy strncasecmp strndup strnlen strrchr strsep strspn strstr strtod strtol strtold strtoq unsetenv uselocale utime vasprintf getpeereid sysctl swapctl malloc_trim])
+AC_CHECK_FUNCS([asprintf atexit closefrom dup2 eaccess endpwent euidaccess ffsll ftruncate getcwd gethostbyname gethostname getloadavg gettimeofday glob ioperm inet_ntoa isascii memchr memmove memset mkdir mkdtemp munmap newlocale pipe2 ppoll putenv re_comp regcomp select setenv socket strcasecmp strcasestr strchr strcspn strdup strerror strftime_l strlcat strlcpy strncasecmp strndup strnlen strptime_l strrchr strsep strspn strstr strtod strtol strtold strtoq unsetenv uselocale utime vasprintf getpeereid sysctl swapctl malloc_trim])
 
 AC_MSG_CHECKING(for htonll)
 AC_LINK_IFELSE(
@@ -949,9 +943,9 @@ AC_SUBST(PBX_DYNAMIC_LIST)
 LDFLAGS=${old_LDFLAGS}
 rm -f conftest.dynamics
 
-AC_CHECK_HEADER([sys/poll.h],
+AC_CHECK_HEADER([poll.h],
    [HAS_POLL=1]
-   AC_DEFINE([HAVE_SYS_POLL_H], 1, [Define to 1 if your system has working sys/poll.h]),
+   AC_DEFINE([HAVE_SYS_POLL_H], 1, [Define to 1 if your system has working poll.h]),
    )
 
 AC_ARG_ENABLE([internal-poll],
